Secrets Store documentation (#567)
EGI-Federation/documentation: content/en/users/security/secrets-store/_index.md

---
title: EGI Secrets Store
linkTitle: Secrets Store
type: docs
weight: 60
description: >
  Managing secrets in the EGI infrastructure
---

## What is it?

Your applications and services may need different secrets (credentials, tokens,
passwords, etc.) during deployment and operation. These secrets are often
stored as clear texts in code repositories or configuration files, which
poses security risks. Furthermore, the secrets stored in files are static and
difficult to change (rotate).

The EGI Secrets Store helps you to
**securely store, retrieve, and rotate credentials for your services**.

The main features of EGI Secrets Store:

> - Simplifies, standardizes and secures the way users store secrets,

Reviewer: ```suggestion
- Simplify, standardize and secure the way users store secrets,
```

Author: Does not fit with the rest of the bullet points. Either all verbs/actions or all nouns/features.

- Secure storage of secrets, as they are encrypted both in transit and at rest.
- Makes it easy to rotate secrets, while administrators can track usage and
  manage access.

> - Integration with [EGI Check-in](../../aai/check-in/) allows access with
>   community credentials, no additional registration needed.

Reviewer: ```suggestion
- Integration with [EGI Check-in](../../aai/check-in/) allowing access with
  home organisation credentials, no additional registration needed.
```

Author: Fixed

> - High-availability ensures this critical service is always accessible to
>   users, applications and other services built on top of it.

Reviewer: ```suggestion
- High-availability ensuring the service is always accessible to
  users, applications and other services relying on it.
```

Author: Use of gerund does not fit here (does not match the rest of the bullet points). Dropped "critical".

> - Service is easy to maintain, upgrade, and extend, as it is based on
>   well-known open-source software, with many client tools and libraries that
>   have strong community support.

Reviewer: ```suggestion
- Service is easy to maintain, upgrade, and extend. It is based on
  well-known open-source software, with many client tools and libraries
  having a strong community support.
```

Author: All other bullets are single sentences, this should also be one. Gerund use for informal texts, does not fit here

## Concepts

**Secret objects** (or secrets, for short) in EGI Secrets Store are identified
by their paths, like files on disk. Each user has a private **secret space**

> for storing his/her secret objects, and cannot see secrets of other users.

Reviewer: ```suggestion
for storing their secret objects, and cannot see secrets of other users.
```

Author: Fixed

Each secret object may contain several **secret values**, and each value is
identified by its **key** (name).

> {{% alert title="Example" color="info" %}} A secret object for your service
> may contain several different passwords for different components: database,
> message queue, cache, storage system, etc. Each of these passwords will be a
> distinct secret value in your secret object.{{% /alert %}}

Reviewer: ```suggestion
{{% alert title="Example" color="info" %}} A secret object for a service
may contain several different passwords for different components: database,
message queue, cache, storage system, etc. Each of these passwords will be a
distinct secret value in the secret object.{{% /alert %}}
```

Author: No. Our style guide recommends 2nd person. Not a user, you. Not a service, your service.
https://docs.egi.eu/about/contributing/style/#write-in-second-person
